ts/parsers/elementparser.ts

```diff
@@ -7,49 +7,134 @@ import * as isEmpty from "lodash.isempty";
 /**
  * Entry point for all Parsers
  */
+const SUPPORTED_PLACEHOLDERS = ["body", "ctrTitle", "pic", "subTitle", "tbl", "title"];
 class PowerpointElementParser {
     private element;
 
-    public getProcessedElement(rawElement, slideRelationships): PowerpointElement {
+    public isNonSupportedPlaceholder() {
+        if (checkPath(this.element, '["p:nvSpPr"][0]["p:nvPr"][0]["p:ph"][0]["$"]["type"]')) {
+            const type = getValueAtPath(
+                this.element,
+                '["p:nvSpPr"][0]["p:nvPr"][0]["p:ph"][0]["$"]["type"]'
+            );
+            if (!SUPPORTED_PLACEHOLDERS.includes(type)) {
+                return true;
+            }
+        }
+
+        return false;
+    }
+
+    public isPlaceholderListElement(slideLayoutTables): boolean {
+        if (checkPath(this.element, '["p:nvSpPr"][0]["p:nvPr"][0]["p:ph"][0]["$"]["idx"]')) {
+            const placeholderIdx = getValueAtPath(
+                this.element,
+                '["p:nvSpPr"][0]["p:nvPr"][0]["p:ph"][0]["$"]["idx"]'
+            );
+            let slideLayoutSpNode = undefined;
+            slideLayoutSpNode = slideLayoutTables["idxTable"][placeholderIdx];
+
+            return (
+                slideLayoutSpNode !== undefined &&
+                checkPath(slideLayoutSpNode, '["p:txBody"][0]["a:p"][0]["a:pPr"][0]["$"]["lvl"]') &&
+                checkPath(slideLayoutSpNode, '["p:txBody"][0]["a:lstStyle"][0]["a:lvl1pPr"][0]["a:buNone"]') === false
+            );
+        }
+
+        return false;
+    }
+
+    public getLayoutSpNodes(slideLayoutTables, slideMasterTables) {
+        const idx =
+            getAttributeByPath(this.element, ["p:nvSpPr", "p:nvPr", "p:ph"]) === undefined
+                ? undefined
+                : getAttributeByPath(this.element, ["p:nvSpPr", "p:nvPr", "p:ph", "$", "idx"]);
+        const type =
+            getAttributeByPath(this.element, ["p:nvSpPr", "p:nvPr", "p:ph"]) === undefined
+                ? undefined
+                : getAttributeByPath(this.element, ["p:nvSpPr", "p:nvPr", "p:ph", "$", "type"]);
```

Comment:
```suggestion
        let idx = undefined;
        let type = undefined;
        
        if (getAttributeByPath(this.element, ["p:nvSpPr", "p:nvPr", "p:ph"]) !== undefined) {
            idx = getAttributeByPath(this.element, ["p:nvSpPr", "p:nvPr", "p:ph", "$", "idx"]);
            type = getAttributeByPath(this.element, ["p:nvSpPr", "p:nvPr", "p:ph", "$", "type"]);
        }
```

ternary's can be gotten rid off (saity check by change before comitting directly, I am reviewing this with a lot distractions)

Reply:
check was not efficient, removed ternary, getting values directly now, if we don't find it we will have undefined, and checks for undefined values is already there in the code below